File: src/main/java/net/igneo/icv/entity/divineLightningRod/DivineLightningRodModel.java
package net.igneo.icv.entity.divineLightningRod;

import net.igneo.icv.ICV;
import net.minecraft.resources.ResourceLocation;
import software.bernie.geckolib.model.GeoModel;

public class DivineLightningRodModel extends GeoModel<DivineLightningRodEntity> {
  private final ResourceLocation model = new ResourceLocation(ICV.MOD_ID, "geo/placeholder.geo.json");
  private final ResourceLocation texture = new ResourceLocation(ICV.MOD_ID, "textures/entity/placeholder.png");
  private final ResourceLocation animations = new ResourceLocation(ICV.MOD_ID, "animations/placeholder.animation.json");
  @Override
  public ResourceLocation getModelResource(DivineLightningRodEntity animatable) {
    return model;
  }
  
  @Override
  public ResourceLocation getTextureResource(DivineLightningRodEntity animatable) {
    return texture;
  }
  
  @Override
  public ResourceLocation getAnimationResource(DivineLightningRodEntity animatable) {
    return animations;
  }
}

File: src/main/java/net/igneo/icv/entity/helmet/divineLightningRod/DivineLightningRodModel.kt
package net.igneo.icv.entity.helmet.divineLightningRod

import net.igneo.icv.ICV
import net.minecraft.resources.ResourceLocation
import software.bernie.geckolib.model.GeoModel

class DivineLightningRodModel : GeoModel<DivineLightningRodEntity?>() {
    private val model = ResourceLocation(ICV.MOD_ID, "geo/placeholder.geo.json")
    private val texture = ResourceLocation(ICV.MOD_ID, "textures/entity/placeholder.png")
    private val animations = ResourceLocation(ICV.MOD_ID, "animations/placeholder.animation.json")

    override fun getModelResource(animatable: DivineLightningRodEntity?): ResourceLocation {
        return model
    }

    override fun getTextureResource(animatable: DivineLightningRodEntity?): ResourceLocation {
        return texture
    }

    override fun getAnimationResource(animatable: DivineLightningRodEntity?): ResourceLocation {
        return animations
    }
}
